blobs.go

```diff
@@ -92,6 +92,10 @@ type BlobStatter interface {
 	Stat(ctx context.Context, dgst digest.Digest) (Descriptor, error)
 }
 
+type BlobAccessFactory interface {
+	BlobAccessController(svc BlobDescriptorService) BlobDescriptorService
```

Comment:
> Clear removes access to a blob from a repository. Calling this should make Stat return false, but it doesn't remove any blobs from the blob store.

Yes, it does not remove any blobs. Since we can't prevent the creation of links we need to remove them as it makes `linkedBlobStore`. Otherwise, storage will accumulate the garbage that we have no way to clean.

> Does your custom BlobDescriptorService implement Stat based on it's metadata implementation?

Our Stat implementation uses BlobStatter directly without `linkedBlobStatter`.